### PR TITLE
OHRM5X-2335: Group By Fix

### PR DESCRIPTION
**Pull request by devishke-orange**
PR contains:
- OHRM5X-2335: Fix Group By Installer Failure

[Research](https://docs.google.com/document/d/1hdc883exY_0rYL-waHZMn9RAA6bJ86PMa1kYkCdaGjU/edit#heading=h.fpex38c77zzb)

### CODE REVIEW

**Comment by devishke-orange** [comment]
@RajithaKumara shall we change our main test.yml to use `mysql 5.7` instead of `mariadb 10.3`? If group by errors  are present while developing it will be detected sooner than waiting for scheduled tests.

File: src/plugins/orangehrmLeavePlugin/Dao/LeaveRequestDao.php [hunk 1]
```diff
@@ -559,6 +559,9 @@ private function getLeaveRequestsPaginator(
         $q->addOrderBy('employee.lastName', ListSorter::ASCENDING)
             ->addOrderBy('employee.firstName', ListSorter::ASCENDING);
 
+        $q->addSelect('leaveRequest.id');
```
Comment (RajithaKumara) [inline src/plugins/orangehrmLeavePlugin/Dao/LeaveRequestDao.php:562]: *

File: src/plugins/orangehrmPimPlugin/Dto/EmployeeAllowedLanguageSearchFilterParams.php [hunk 2]
```diff
@@ -22,7 +22,7 @@
 
 class EmployeeAllowedLanguageSearchFilterParams extends FilterParams
 {
-    public const ALLOWED_SORT_FIELDS = ['l.name'];
+    public const ALLOWED_SORT_FIELDS = ['language.name'];
```
Comment (RajithaKumara) [inline src/plugins/orangehrmPimPlugin/Dto/EmployeeAllowedLanguageSearchFilterParams.php:25]: *

**Review by RajithaKumara — APPROVED** [review]

File: src/plugins/orangehrmLeavePlugin/Dto/LeaveRequestSearchFilterParams.php [hunk 3]
```diff
@@ -29,7 +29,7 @@ class LeaveRequestSearchFilterParams extends DateRangeSearchFilterParams
     use SubunitIdChainTrait;
     use LeaveRequestServiceTrait;
 
-    public const ALLOWED_SORT_FIELDS = ['leave.date'];
+    public const ALLOWED_SORT_FIELDS = ['minDate'];
```
[the remark this answers is not in the document]
Comment (devishke-orange) [reply src/plugins/orangehrmLeavePlugin/Dto/LeaveRequestSearchFilterParams.php]: changed

**Pull request merged**